include/picongpu/particles/atomicPhysics2/initElectrons/Inelastic2BodyCollisionFromCoMoving.hpp

```diff
@@ -0,0 +1,278 @@
+
+            //      Lorentz transformation from IonSystem to LabSystem
+            // square of original momentum of the ion in Lab-System
+            // UNIT_MASS^2 * UNIT_LENGTH^2 / UNIT_TIME^2, not weighted
+            float_64 const momentumSquaredIon_LabSystem
+                = static_cast<float_64>(pmacc::math::l2norm2(ion[momentum_]) / (ion[weighting_] * ion[weighting_]));
+
+            // unitless
+            float3_64 betaIonSystem;
+            // unitless
+            float_64 normSquaredBetaIon_LabSystem;
+
+            if(momentumSquaredIon_LabSystem == 0._X)
+            {
+                /// @note momentum is always 3 dimensional in picongpu
+                betaIonSystem = float3_64(0., 0., 0.);
+                normSquaredBetaIon_LabSystem = 0.;
+            }
+            else
+            {
+                // square of original beta of the ion in Lab-System
+                // unitless
+                // beta^2 = 1/(1 + (m^2*c^2)/p^2)
+                //  unitless + (UNIT_MASS^2 * UNIT_SPEED^2)(not weighted)
+                //  /( UNIT_MASS^2 * UNIT_SPEED^2)(not weighted) = unitless
+                normSquaredBetaIon_LabSystem = 1. / (1. + mSquaredCSquaredIon / momentumSquaredIon_LabSystem);
+
+                betaIonSystem =
+                    // magnitude
+                    math::sqrt(normSquaredBetaIon_LabSystem)
+                    // direction
+                    * static_cast<float3_64>(ion[momentum_] / pmacc::math::l2norm(ion[momentum_]));
+            }
+
+            // unitless
+            auto beta = MatrixVector(betaIonSystem);
+
+            float_64 const gammaIonSystem = math::sqrt(momentumSquaredIon_LabSystem / mSquaredCSquaredIon + 1.);
+
+            // lower 3x3 block of Lorentz transformation matrix for transformation from
+            // Ion-System to Lab-System
+            Matrix_3x3 lorentzMatrix;
+            fillLorentzMatrix(beta, gammaIonSystem, normSquaredBetaIon_LabSystem, lorentzMatrix);
+
+            //      space components of Lorentz boost
+            MatrixVector momentumStarElectron_LabSystem;
+            lorentzMatrix.mMul(momentumStarElectron_IonSystem, momentumStarElectron_LabSystem);
+
+            //      calculate momenta after collision
+            //          ion
+            // UNIT_MASS * UNIT_LENGTH/UNIT_TIME, weighted
+            MatrixVector momentumStarIon_LabSystem = (momentumStarElectron_LabSystem.sMul(-1.))
+                + (beta.sMul(gammaIonSystem * gammaStarIon_IonSystem * mcIon));
+
+            //          electron
+            // UNIT_MASS * UNIT_LENGTH / UNIT_TIME, weighted
+            momentumStarElectron_LabSystem = momentumStarElectron_LabSystem
+                + (beta.sMul(gammaIonSystem * gammaStarElectron_IonSystem * mcElectron));
+
+            // set to particle
+#pragma unroll
+            for(uint32_t i = 0u; i < picongpu::simDim; i++)
```

Comment:
momentum has always 3 components! even in a 2D siulation

Reply:
fixed